R/get.R

```diff
@@ -217,3 +217,53 @@ get_seeding_time <- function(delays, generation_time, rt = rt_opts()) {
   }
   return(max(round(seeding_time), 1))
 }
+
+#' Get posterior samples from a fitted model
+#'
+#' @description `r lifecycle::badge("stable")`
+#' Extracts posterior samples from a fitted model, combining all parameters
+#' into a single data.table with dates and metadata.
+#'
+#' @param object A fitted model object (e.g., from `estimate_infections()`)
+#' @param ... Additional arguments (currently unused)
+#'
+#' @return A `data.table` with columns: date, variable, strat, sample, time,
+#'   value, type. Contains all posterior samples for all parameters.
+#'
+#' @export
+#' @examples
+#' \dontrun{
+#' # After fitting a model
+#' samples <- get_samples(fit)
+#' # Filter to specific parameters
+#' R_samples <- samples[variable == "R"]
+#' }
+get_samples <- function(object, ...) {
+  UseMethod("get_samples")
+}
+
+#' @rdname get_samples
+#' @export
+get_samples.estimate_infections <- function(object, ...) {
```

Comment:
I assume we want a new issue to make a method for this for epinow and for other functions where it makes sense or we don't need it to be s3.